nova_vm/src/ecmascript/builtins/data_view/abstract_operations.rs

```diff
@@ -143,3 +154,147 @@ pub(crate) fn is_view_out_of_bounds(
+    let view_offset = view.byte_offset(agent);
+
+    // 6. Let viewRecord be MakeDataViewWithBufferWitnessRecord(view, unordered).
+    let view_record = make_data_view_with_buffer_witness_record(agent, view, Ordering::Unordered);
+
+    // 7. NOTE: Bounds checking is not a synchronizing operation when view's backing buffer is a growable SharedArrayBuffer.
+    // 8. If IsViewOutOfBounds(viewRecord) is true, throw a TypeError exception.
+    if is_view_out_of_bounds(agent, &view_record) {
+        return Err(agent.throw_exception_with_static_message(
+            ExceptionType::TypeError,
+            "DataView is out of bounds",
+        ));
+    }
+
+    // 9. Let viewSize be GetViewByteLength(viewRecord).
+    let view_size = get_view_byte_length(agent, &view_record);
+
+    // 10. Let elementSize be the Element Size value specified in Table 69 for Element Type type.
+    let element_size = size_of::<T>();
+
+    // 11. If getIndex + elementSize > viewSize, throw a RangeError exception.
+    if get_index + element_size > view_size {
+        return Err(agent.throw_exception_with_static_message(
+            ExceptionType::RangeError,
+            "Index out of bounds",
+        ));
+    }
+
+    // 12. Let bufferIndex be getIndex + viewOffset.
+    let buffer_index = get_index + view_offset;
+
+    // 13. Return GetValueFromBuffer(view.[[ViewedArrayBuffer]], bufferIndex, type, false, unordered, isLittleEndian).
+    Ok(get_value_from_buffer::<T>(
+        agent,
+        view.get_viewed_array_buffer(agent),
+        buffer_index,
+        false,
+        Ordering::Unordered,
+        Some(is_little_endian),
+    ))
+}
+
+/// ### [25.3.1.6 SetViewValue ( view, requestIndex, isLittleEndian, type, value )](https://tc39.es/ecma262/#sec-setviewvalue)
+///
+/// The abstract operation SetViewValue takes arguments view (an ECMAScript
+/// language value), requestIndex (an ECMAScript language value), isLittleEndian
+/// (an ECMAScript language value), type (a TypedArray element type), and value
+/// (an ECMAScript language value) and returns either a normal completion
+/// containing undefined or a throw completion. It is used by functions on
+/// DataView instances to store values into the view's buffer.
+pub(crate) fn set_view_value<T: Viewable>(
+    agent: &mut Agent,
+    mut gc: GcScope<'_, '_>,
+    view: Value,
+    request_index: Value,
+    // 6. Set isLittleEndian to ToBoolean(isLittleEndian).
+    is_little_endian: bool,
+    value: Value,
+) -> JsResult<Value> {
+    // 1. Perform ? RequireInternalSlot(view, [[DataView]]).
+    // 2. Assert: view has a [[ViewedArrayBuffer]] internal slot.
+    let view = require_internal_slot_data_view(agent, view)?;
+
+    // 3. Let getIndex be ? ToIndex(requestIndex).
+    let get_index = to_index(agent, gc.reborrow(), request_index)? as usize;
+
+    // 4. If IsBigIntElementType(type) is true, let numberValue be ? ToBigInt(value).
+    let number_value = if T::IS_BIGINT {
```

Comment:
**suggestion:** I think if we add a `T::NumberValueType` which is `Number` for non-64-bit viewables and `BigInt` for the rest, then we can here do `let number_value: T::NumberValueType = if T::IS_BIGINT`, and then the `from_xe_value` variants won't need `gc` anymore as they're always converting from `Number` to  native type or from `BigInt` to native type.

Reply:
I'll leave that for a future improvement of the `Viewable` trait. We'll also need a `IS_CLAMPED` trait const or something for when we implement clamped arrays.